spec/controllers/verify/sessions_controller_spec.rb

```diff
@@ -32,44 +34,121 @@
+        it 'shows normal form with error' do
+          post :create, profile: user_attrs.merge(ssn: '')
+
+          expect(response).to_not redirect_to(verify_session_dupe_path)
+          expect(response.body).to match t('errors.messages.blank')
+        end
+      end
+
+      context 'missing fields' do
+        it 'checks for required fields' do
+          partial_attrs = user_attrs.dup
+          partial_attrs.delete :first_name
 
-        post :create, profile: user_attrs.merge(ssn: '1234')
+          post :create, profile: partial_attrs
 
-        expect(response).to redirect_to(verify_session_dupe_path)
-        expect(flash[:error]).to match t('idv.errors.duplicate_ssn')
+          expect(response).to render_template(:new)
+          expect(response.body).to match t('errors.messages.blank')
+        end
       end
 
-      it 'shows normal form with error on empty SSN' do
-        post :create, profile: user_attrs.merge(ssn: '')
+      context 'un-resolvable attributes' do
+        let(:bad_attrs) { user_attrs.dup.merge(first_name: 'Bad') }
+
+        it 're-renders form' do
+          post :create, profile: bad_attrs
+
+          expect(response).to render_template(:new)
+        end
+
+        it 'creates analytics event' do
+          post :create, profile: bad_attrs
+
+          result = {
+            success: false,
+            idv_attempts_exceeded: false
+          }
+
+          expect(@analytics).to have_received(:track_event).with(Analytics::IDV_INITIAL, result)
+        end
+      end
+
+      context 'success' do
+        it 'creates vendor artifacts' do
+          post :create, profile: user_attrs
+
+          resolution = idv_session.resolution
+          expect(resolution).to be_a Proofer::Resolution
+          expect(resolution.success).to eq true
+
+          applicant = idv_session.applicant
+          expect(applicant).to be_a Proofer::Applicant
+        end
+
+        it 'creates analytics event' do
+          post :create, profile: user_attrs
+
+          result = {
+            success: true,
+            idv_attempts_exceeded: false
+          }
 
-        expect(response).to_not redirect_to(verify_session_dupe_path)
-        expect(response.body).to match t('errors.messages.blank')
+          expect(@analytics).to have_received(:track_event).with(Analytics::IDV_INITIAL, result)
+        end
       end
 
-      it 'checks for required fields' do
-        partial_attrs = user_attrs.dup
-        partial_attrs.delete :first_name
+      context 'max attempts exceeded' do
+        before do
+          user.idv_attempts = 3
+          user.idv_attempted_at = Time.zone.now
+        end
 
-        post :create, profile: partial_attrs
+        it 'redirects to fail' do
+          post :create, profile: user_attrs
 
-        expect(response).to render_template(:new)
-        expect(response.body).to match t('errors.messages.blank')
+          expect(@analytics).to_not have_received(:track_event)
```

Comment:
There is no existing analytics event for max attempts. I was trying not to create new features here, just refactor to split the existing feature across individual steps.

It's fine to create a new analytics event to track max attempts exceeded, in which case this will fail and need to be changed to reflect that new feature.